packages/blade/src/components/Drawer/Drawer.web.tsx

```diff
@@ -0,0 +1,225 @@
+import {
+  FloatingFocusManager,
+  FloatingOverlay,
+  FloatingPortal,
+  useFloating,
+} from '@floating-ui/react';
+import type { CSSProperties } from 'react';
+import React from 'react';
+import styled from 'styled-components';
+import usePresence from 'use-presence';
+import { drawerComponentIds } from './drawerComponentIds';
+import { DrawerContext } from './DrawerContext';
+import type { DrawerProps } from './types';
+import { Box } from '~components/Box';
+import BaseBox from '~components/Box/BaseBox';
+import { castWebType, makeMotionTime, useTheme } from '~utils';
+import { assignWithoutSideEffects } from '~utils/assignWithoutSideEffects';
+import { componentZIndices } from '~utils/componentZIndices';
+import { useDrawerStack } from '~components/Drawer/StackProvider';
+import { makeAccessible } from '~utils/makeAccessible';
+import { metaAttribute, MetaConstants } from '~utils/metaAttribute';
+import { useId } from '~utils/useId';
+import { useVerifyAllowedChildren } from '~utils/useVerifyAllowedChildren';
+
+const SHOW_DRAWER = 'show-drawer';
+
+const AnimatedDrawerContainer = styled(BaseBox)<{
+  isFirstDrawerInStack: boolean;
+  isVisible: boolean;
+}>(({ theme, isFirstDrawerInStack, isVisible }) => {
+  const entranceTransition: CSSProperties['transition'] = `all ${castWebType(
+    castWebType(makeMotionTime(theme.motion.duration.gentle)),
+  )} ${castWebType(theme.motion.easing.entrance.revealing)}`;
+
+  const exitTransition: CSSProperties['transition'] = `all
+  ${castWebType(makeMotionTime(theme.motion.duration.xmoderate))}
+  ${castWebType(theme.motion.easing.exit.revealing)}`;
+
+  return {
+    opacity: isVisible ? 1 : 0,
+    transform: isVisible
+      ? isFirstDrawerInStack
+        ? 'translateX(calc(-100% - 16px))'
+        : 'translateX(-100%)'
+      : 'translateX(0%)',
+    transition: isVisible ? entranceTransition : exitTransition,
+    animationFillMode: 'initial',
+  };
+});
+
+const DrawerOverlay = styled(FloatingOverlay)(({ theme }) => {
+  return {
+    opacity: 0,
+    transition: `opacity
+      ${makeMotionTime(theme.motion.duration.xmoderate)}
+      ${castWebType(theme.motion.easing.exit.revealing)}`,
+    backgroundColor: theme.colors.overlay.background.subtle,
+
+    [`&.${SHOW_DRAWER}`]: {
+      opacity: 1,
+      transition: `opacity ${makeMotionTime(theme.motion.duration.gentle)} ${castWebType(
+        theme.motion.easing.entrance.revealing,
+      )}`,
+    },
+  };
+});
+
+const _Drawer = ({
+  isOpen,
+  onDismiss,
+  zIndex = componentZIndices.drawer,
+  children,
+  accessibilityLabel,
+  showOverlay = true,
+  initialFocusRef,
+  testID,
+}: DrawerProps): React.ReactElement => {
+  const closeButtonRef = React.useRef<HTMLDivElement>(null);
+  const [zIndexState, setZIndexState] = React.useState<number>(zIndex);
+
+  useVerifyAllowedChildren({
+    children,
+    componentName: 'Drawer',
+    allowedComponents: [drawerComponentIds.DrawerHeader, drawerComponentIds.DrawerBody],
+  });
```

Comment:
Ideally this can be wrapped in __DEV__ too
```suggestion
if (__DEV__) {
  useVerifyAllowedChildren({
    children,
    componentName: 'Drawer',
    allowedComponents: [drawerComponentIds.DrawerHeader, drawerComponentIds.DrawerBody],
  });
 }
```

Reply:
That breaks the react hooks rule so we'll get errors in development no? We follow similar pattern at other places

<img width="893" alt="image" src="https://github.com/razorpay/blade/assets/30949385/dfb99322-e6ab-4f24-b981-3681619fbbac">

Reply:
Oh! yes, correct.